codec-http2/src/test/java/io/netty/handler/codec/http2/Http2FrameCodecTest.java

```diff
@@ -0,0 +1,484 @@
+        Http2ResetFrame actualRst = inboundHandler.readInboundMessagesAndEvents();
+        assertEquals(expectedRst, actualRst);
+
+        Http2StreamClosedEvent closedEvent = inboundHandler.readInboundMessagesAndEvents();
+        assertNotNull(closedEvent);
+        assertEquals(stream.id(), closedEvent.streamId());
+
+        assertNull(inboundHandler.readInboundMessagesAndEvents());
+    }
+
+    @Test
+    public void sendGoAway() throws Exception {
+        frameListener.onHeadersRead(http2HandlerCtx, 3, request, 31, false);
+
+        Http2Stream stream = framingCodec.connectionHandler().connection().stream(3);
+        assertNotNull(stream);
+        assertEquals(State.OPEN, stream.state());
+
+        ByteBuf debugData = bb("debug");
+        Http2GoAwayFrame goAwayFrame = new DefaultHttp2GoAwayFrame(Http2Error.NO_ERROR.code(), debugData);
+        goAwayFrame.setExtraStreamIds(2);
+
+        inboundHandler.writeOutbound(goAwayFrame);
+        verify(frameWriter).writeGoAway(
+                eq(http2HandlerCtx), eq(7), eq(Http2Error.NO_ERROR.code()), eq(debugData), anyChannelPromise());
+        assertEquals(State.OPEN, stream.state());
+        assertTrue(channel.isActive());
+    }
+
+    @Test
+    public void receiveGoaway() throws Exception {
+        frameListener.onGoAwayRead(http2HandlerCtx, 2, Http2Error.NO_ERROR.code(), bb("foo"));
+
+        Http2GoAwayFrame expectedFrame = new DefaultHttp2GoAwayFrame(2, Http2Error.NO_ERROR.code(), bb("foo"));
+        Http2GoAwayFrame actualFrame = inboundHandler.readInbound();
+
+        assertEquals(expectedFrame, actualFrame);
+        assertNull(inboundHandler.readInbound());
+    }
+
+    @Test
+    public void unknownFrameTypeShouldThrow() throws Exception {
+        class UnknownHttp2Frame implements Http2Frame {
+            @Override
+            public String name() {
+                return "UNKNOWN";
+            }
+        }
+
+        ChannelFuture f = channel.write(new UnknownHttp2Frame());
+        f.await();
+        assertTrue(f.isDone());
+        assertFalse(f.isSuccess());
+        assertThat(f.cause(), instanceOf(UnsupportedMessageTypeException.class));
+    }
+
+    @Test
+    public void incomingStreamActiveShouldFireUserEvent() throws Exception {
+        frameListener.onHeadersRead(http2HandlerCtx, 3, request, 31, false);
+
+        Http2Stream stream = framingCodec.connectionHandler().connection().stream(3);
+        assertNotNull(stream);
+
+        Http2HeadersFrame frame = inboundHandler.readInbound();
+        assertNotNull(frame);
+
+        Http2StreamActiveEvent streamActiveEvent = inboundHandler.readUserEvent();
+        assertEquals(stream.id(), streamActiveEvent.streamId());
+
+        assertNull(inboundHandler.readInbound());
+        assertNull(inboundHandler.readUserEvent());
+    }
+
+    @Test
+    public void outgoingStreamActiveShouldNotFireUserEvent() throws Exception {
```

Comment:
this test seems to be verifying that a `Http2StreamActiveEvent` is fired ... consider clarifying the test name.